src/lib/GenericTraversal.hs

```diff
@@ -106,13 +105,9 @@ instance GenericallyTraversableE Atom where
 
 instance GenericallyTraversableE Block where
   traverseGenericE (Block _ decls result) = do
-    Abs decls' (PairE ty result') <-
-      buildScoped $ traverseDeclNest decls do
-        result' <- traverseAtom result
-        resultTy <- getType result'
-        return $ PairE resultTy result'
-    ty' <- liftHoistExcept $ hoist decls' ty
-    return $ Block (BlockAnn ty') decls' result'
+    -- Recompute the effects for the result Block, because the
+    -- traversal may change them
```

Comment:
There's no effect recomputation visible here now. It's all in `buildBlock`, so I find this a bit confusing

Reply:
Comment removed.